File: lists/pig_latin.py


def piggy(s):
    arr = s.split(' ')
    vowels = 'aeiou'
    result = []

    for word in arr:
        cap = False 
        if len(word) > 3:
            if word[0] == word[0].upper():
                cap = True 
            pre = ''
            x = 0
            if word[x] not in vowels:
                pre += word[x]
                x += 1
            else:
                while word[x] in vowels:
                    pre += word[x]
                    x += 1
            temp = word[x:] + pre + 'ay'
            if cap:
                temp = temp.lower().capitalize()
            result.append(temp)
        else:
            result.append(word)

    return ' '.join(result)

text = 'this is a test'

print(piggy(text))

File: lists/pig_latin.js


const piggy = str => {
    let arr = str.split(' ');
    let result = [];

    for (word in arr) {
        let isCap = false;
        let vowels = 'aeiou';

        if (arr[word].length > 3) {
            let x = 0, pre = '';
            if (arr[word][0] == arr[word][0].toUpperCase()) isCap = true;
            for (l in arr[word]) {
                if (!vowels.includes(arr[word][l])) {
                    console.log('not in vowels');
                    pre += arr[word][l];
                    x++;
                } else {
                    while (vowels.includes(arr[word][x])) {
                        console.log('in vowels');
                        pre += arr[word][x];
                        x++;
                    }
                }
                let temp = arr[word].slice(x) + pre + 'ay';

                if (isCap) {
                    temp = temp.toLowerCase()
                    temp = temp[0].toUpperCase() + temp.slice(1);
                }
                result.push(temp);
            }
        } else {
            result.push(arr[word]);
        }
    }
    return result.join(' ');
}

let s = 'This is a test';

console.log(piggy(s));